putting custom mobile ui guide into single page format

chrisdmills-okta opened the pull request:
<!-- PLEASE REMEMBER THAT THIS A PUBLIC REPO AND ANY PR AND SUBSEQUENT DISCUSSION WILL BE VISIBLE TO ANYONE AND EVERYONE -->

## Description:
- **What's changed?** I put https://developer.okta.com/docs/guides/build-custom-ui-mobile/react-native/before-you-begin/ into single page article format.
- **Is this PR related to a Monolith release?** No
- Redirects tested on staging, 11/17/21. Redirects work as expected.

### Resolves:

* [OKTA-438727](https://oktainc.atlassian.net/browse/OKTA-438727)

vanngo-okta commented on packages/@okta/vuepress-site/docs/guides/build-custom-ui-mobile/main/index.md:
```diff
@@ -0,0 +1,76 @@
+
+Every authentication transaction starts with primary authentication, which validates a user's password. Password Policy, MFA Policy, and Sign-On Policy are evaluated during primary authentication to determine if the user's password is expired, a factor should be enrolled, or additional verification is required. The [transaction state](/docs/api/resources/authn/#transaction-state) of the response  ...
+
+> **Note:** Custom sign-in works only with [Org MFA](/docs/guides/mfa/sms/set-up-org/). This means that, before you exchange the session token for an access token, you must make sure the application has disabled [App-Level MFA](https://help.okta.com/okta_help.htm?id=ext_MFA_App_Level).
```
```suggestion
> **Note:** Custom sign-in only works with [Org MFA](/docs/guides/mfa/sms/set-up-org/). This means that before you exchange the session token for an access token, you must ensure that [App-Level MFA](https://help.okta.com/okta_help.htm?id=ext_MFA_App_Level) is disabled for the application.
```

vanngo-okta commented on packages/@okta/vuepress-site/docs/guides/build-custom-ui-mobile/main/index.md:
```diff
@@ -0,0 +1,76 @@
+
+You should now understand how to build a custom UI in your mobile application.
+
+When a user signs in, their profile information (stored in Okta) is made available to your application. It's common to use this information to update your app's UI. See [Get info about the user](/docs/guides/sign-into-mobile-app/get-user-info/) for the steps.
```
```suggestion
When a user signs in, their profile information (stored in Okta) is made available to your application. Use this information to personalize your app's UI for the user. See [Get info about the user](/docs/guides/sign-into-mobile-app/-/get-user-info/) for details.
```

vanngo-okta commented on packages/@okta/vuepress-site/conductor.yml:
```diff
@@ -2523,7 +2523,78 @@ redirects:
   - from: /docs/guides/set-up-self-service-registration/next-steps/index.html
     to: /docs/guides/set-up-self-service-registration/main/
-
+  - from: /docs/guides/build-custom-ui-mobile/react-native/before-you-begin/
```
```suggestion
  - from: /docs/guides/build-custom-ui-mobile/react-native/before-you-begin
```

vanngo-okta replied:
Looks like you've already added /index.html entry (below) 👍

vanngo-okta commented on packages/@okta/vuepress-site/conductor.yml:
```diff
@@ -2523,7 +2523,78 @@ redirects:
-
+  - from: /docs/guides/build-custom-ui-mobile/react-native/before-you-begin/
+    to: /docs/guides/build-custom-ui-mobile/react-native/main/
+  - from: /docs/guides/build-custom-ui-mobile/react-native/create-an-okta-application/
```
```suggestion
  - from: /docs/guides/build-custom-ui-mobile/react-native/create-an-okta-application
```
Also add: from: /docs/guides/build-custom-ui-mobile/react-native/create-an-okta-application/index.html

vanngo-okta commented on packages/@okta/vuepress-site/conductor.yml:
```diff
@@ -2523,7 +2523,78 @@ redirects:
+    to: /docs/guides/build-custom-ui-mobile/react-native/main/
+  - from: /docs/guides/build-custom-ui-mobile/react-native/create-an-okta-application/
+    to: /docs/guides/build-custom-ui-mobile/react-native/main/
+  - from: /docs/guides/build-custom-ui-mobile/react-native/configure-packages/
```
```suggestion
  - from: /docs/guides/build-custom-ui-mobile/react-native/configure-packages
```

vanngo-okta commented on packages/@okta/vuepress-site/conductor.yml:
```diff
@@ -2523,7 +2523,78 @@ redirects:
+  - from: /docs/guides/build-custom-ui-mobile/react-native/create-an-okta-application/
+    to: /docs/guides/build-custom-ui-mobile/react-native/main/
+  - from: /docs/guides/build-custom-ui-mobile/react-native/configure-packages/
+    to: /docs/guides/build-custom-ui-mobile/react-native/main/
```
```suggestion
    to: /docs/guides/build-custom-ui-mobile/react-native/main/#add-and-configure-packages
```
Not sure if you want to be specific. This is just a suggestion.

vanngo-okta reviewed (changes requested):
Looks good Chris! This is a good learning experience for me 👍 
Just a few nits.

chrisdmills-okta commented on packages/@okta/vuepress-site/docs/guides/build-custom-ui-mobile/main/index.md:
```diff
@@ -0,0 +1,76 @@
+layout: Guides
+---
+
+You can connect your mobile app to Okta and sign users in by [opening a browser](/docs/guides/sign-into-mobile-app/before-you-begin/), however, if you prefer that your users not leave your app, you need to build a custom sign-in UI with native controls and screens instead. Use this guide to build a customized sign-in experience inside your mobile application.
```
Yeah, this was a language neutral link. I've "fixed" it to what will be correct by the time these PRs are all merged (it would currently be broken). I'm not sure if this is going to work, but we will see what happens when I try it.

chrisdmills-okta commented on packages/@okta/vuepress-site/docs/guides/build-custom-ui-mobile/main/index.md:
```diff
@@ -0,0 +1,76 @@
+
+Next, you need to add Okta to your application by installing and configuring our SDK.
+
+You should already have added and configured packages when you walked through the [Sign users in to your mobile app](/docs/guides/sign-into-mobile-app/android/configure-packages/) guide.
```
Yup, fixed to what it will be.

chrisdmills-okta commented:
@vanngo-okta comments addressed (I think!) Let's see how this one goes.

vanngo-okta commented on packages/@okta/vuepress-site/docs/guides/build-custom-ui-mobile/main/index.md:
```diff
@@ -0,0 +1,76 @@
+**What you need**
+
+* An Okta Developer organization. Don't have one? [Create one for free](https://developer.okta.com/signup)
+* A mobile app that you want to add a custom sign-in UI to
```
```suggestion
* A mobile app with Okta authentication (see [Sign users in to your mobile app](/docs/guides/sign-into-mobile-app/)) that you want to add custom the sign-in UI to
```
Maybe merge these 2 points? The article references [Sign users in to your mobile app] and assumes that Okta integration app exists and that auth is already working.

vanngo-okta commented on packages/@okta/vuepress-site/docs/guides/build-custom-ui-mobile/main/index.md:
```diff
@@ -0,0 +1,76 @@
+
+* An Okta Developer organization. Don't have one? [Create one for free](https://developer.okta.com/signup)
+* A mobile app that you want to add a custom sign-in UI to
+* Knowledge of how Okta mobile auth works (see [Sign users in to your mobile app](/docs/guides/sign-into-mobile-app/))
```
```suggestion
```

vanngo-okta commented on packages/@okta/vuepress-site/docs/guides/build-custom-ui-mobile/main/index.md:
```diff
@@ -0,0 +1,76 @@
+
+Every authentication transaction starts with primary authentication, which validates a user's password. Password Policy, MFA Policy, and Sign-On Policy are evaluated during primary authentication to determine if the user's password is expired, a factor should be enrolled, or additional verification is required. The [transaction state](/docs/api/resources/authn/#transaction-state) of the response  ...
+
+> **Note:** Custom sign-in only works with [Org MFA](/docs/guides/mfa/sms/set-up-org/). This means that before you exchange the session token for an access token, you must ensure that [App-Level MFA](https://help.okta.com/okta_help.htm?id=ext_MFA_App_Level) is disabled for the application.
```
```suggestion
> **Note:** Custom sign-in only works with [Org MFA](/docs/guides/mfa/sms/main/#set-up-your-org-for-mfa). This means that before you exchange the session token for an access token, you must ensure that [App-Level MFA](https://help.okta.com/okta_help.htm?id=ext_MFA_App_Level) is disabled for the application.
```

vanngo-okta reviewed (commented):
Added some conductor.yml updates.
small typo in /ios/handle-response.md: "responsibilty"

chrisdmills-okta commented on packages/@okta/vuepress-site/docs/guides/build-custom-ui-mobile/main/index.md:
```diff
@@ -0,0 +1,76 @@
+
+**Learning outcomes**
+
+* Add the required packages to your application.
```
Sure thing; I agree with you. Updated in my next commit.

chrisdmills-okta commented:
Thanks for the latest comments @vanngo-okta ! I've implemented the changes, and fixed the merge conflicts.

vanngo-okta commented on packages/@okta/vuepress-site/docs/guides/build-custom-ui-mobile/main/index.md:
```diff
@@ -0,0 +1,75 @@
+excerpt: Learn how to build a custom sign-in UI in your mobile app.
+layout: Guides
+---
+
```
The current Nutrition facts template says the stack-selector should go above the summary?

vanngo-okta commented on packages/@okta/vuepress-site/docs/guides/build-custom-ui-mobile/main/index.md:
```diff
@@ -0,0 +1,75 @@
+* factor selection
+* multifactor verification
+
+<StackSelector snippet="primaryauth" />
```
Are we keeping stack selector drop-down menus for each section?

Or should we use `inline` or `noSelector` options in the tags?

`<StackSnippet snippet="inlineSnippet" inline />`
`<StackSelector snippet="sample" noSelector />`

vanngo-okta commented on packages/@okta/vuepress-site/docs/guides/build-custom-ui-mobile/main/index.md:
```diff
@@ -0,0 +1,75 @@
+
+> **Note:** Custom sign-in only works with [Org MFA](/docs/guides/mfa/sms/main/#set-up-your-org-for-mfa). This means that before you exchange the session token for an access token, you must ensure that [App-Level MFA](https://help.okta.com/okta_help.htm?id=ext_MFA_App_Level) is disabled for the application.
+
+<StackSelector snippet="handle-responses" />
```
same comment about hiding stack-selector drop-down menu

vanngo-okta commented on packages/@okta/vuepress-site/docs/guides/build-custom-ui-mobile/main/index.md:
```diff
@@ -0,0 +1,75 @@
+
+## Sample code
+
+<StackSelector snippet="samplecode" />
```
Same comment about hiding stack-selector drop-down menu here. Also, the nutrition fact template might change to put the sample code section back at the top if stack-selectors are involved? I'll discuss with Christine.

vanngo-okta reviewed (commented):
Hi Chris, I have a few questions about the nutrition facts template and how to show stack-selector items in the sample code section. Thanks

chrisdmills-okta commented:
Thanks @vanngo-okta ; your suggestions all looked good, so I committed them. As for the stack selector/nut facts queries. We'll answer those in the meeting later today. I'll hold off on updating this further until then.

chrisdmills-okta commented:
@vanngo-okta I've now updated the stack selectors and nut facts sample code, as agreed in the meeting yesterday. I also fixed the typo you highlighted ages ago...missed that one before now ;-)

I think with that, this one should be good to go

vanngo-okta reviewed (approved):
LGTM 🚀
